protocol/bc/byte32.go

```diff
@@ -0,0 +1,94 @@
+package bc
+
+// TODO(bobg): Both the protobuf version of Hash and the legacy
+// version of AssetID use the methods here. After AssetID gets
+// protobuffified (in terms of Hash, as on the entry-protos4 branch),
+// these methods can be inlined in Hash and byte32 won't be needed.
+
+import (
+	"bytes"
+	"database/sql/driver"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"chain/errors"
+)
+
+type byte32 [32]byte
```

Comment:
Given that (IMO) AssetID should be defined in terms of Hash both in the short term and in the long term, I would really like to see this type eliminated from this PR.